app/common_components/course_viewer/courseViewer.tsx

```diff
@@ -1,74 +1,200 @@
 import LessonList, { ILessonItem } from "./lessonList";
+import LessonCourseToggle from "./lessonCourseToggle";
 
 export interface ICourseViewer {
 	courseName: string
 	courseDescription: string
 	courseThumbnail: string
-	prerequisites?: string[]
+	coursePrerequisites?: string[]
 	lessons: ILessonItem[]
 }
 
 const CourseViewer = ({
   courseName,
   courseDescription,
   courseThumbnail,
-  prerequisites,
+  coursePrerequisites,
   lessons,
 }: ICourseViewer) => {
+  const [ lessonToggle, setLessonToggle ] = useState<boolean>(true);
+  const [ currentLesson, setCurrentLesson ] = useState<ILessonItem | null>(null);
+
+  useEffect(() => {
+    if (lessons.length > 0) {
+      setCurrentLesson(lessons[0]);
+    }
+  }, [lessons]);
+
+  const displayPreRequisites = (coursePrerequisites?: string[]) => {
+    if (!coursePrerequisites) return null;
+    return (
+      <>
+        <h3 className='mb-1 font-medium text-[15px]'>Prerequisites</h3>
+        {coursePrerequisites.map(
+          (coursePrerequisites: string, index: number) => {
+            return (
+              <li className='ml-6 font-light text-sm' key={index}>
+                {coursePrerequisites}
+              </li>
+            );
+          }
+        )}
+      </>
+    );
+  };
+
+  const displayRelatedCourses = () => {
+    return (
+      <div>
+        <h3 className='font-medium mb-0'>Related Courses</h3>
+        <div className='flex mb-4'>
+          <div>
+            <img
+              src={courseThumbnail}
+              alt='Course thumbnail'
+              className='rounded-tl-lg rounded-bl-lg max-h-[150px] '
+            />
+          </div>
+          <div className='bg-[#B96060] rounded-tr-lg rounded-br-lg px-4 py-4 max-h-[150px] w-full'>
+            <h4 className='text-xs mb-2 text-black'>Lesson Name</h4>
+            <p className='text-[10px] leading-none'>
+							Lorem ipsum dolor sit amet consectetur adipisicing
+							elit.
+            </p>
+          </div>
+        </div>
+      </div>
+    );
+  };
+
   return (
     <>
       <div className='flex grow flex-col h-full md:h-auto md:flex-row bg-gradient-to-r from-zinc-600 to-zinc-800'>
-        <div className='left-side md:w-5/12 md:bg-zinc-800 p-5 md:overflow-auto'>
-          <h1 className='mb-0 md:mb-7 text-6xl font-bold'>
-            {courseName}
-          </h1>
-          <div className='hidden md:block'>
-            <LessonList lessons={lessons} />
+        <div className='left-side pt-4 pl-6 pr-6 md:overflow-auto md:w-5/12 md:bg-zinc-800'>
+          <div className='flex items-center justify-between'>
+            <h1 className='mb-0 text-3xl font-medium md:hidden'>
+              {lessonToggle && currentLesson
+                ? currentLesson.lessonName
+                : courseName}
+            </h1>
+            <h1 className='hidden md:block md:mb-7 md:text-5xl'>
+              {courseName}
+            </h1>
+            <div className='min-w-[150px] md:hidden'>
+              {
```

Comment:
Since LessonCourseToggle is being used as a component and not a function, the curly braces here and at Line 108 are unnecessary